### PR TITLE
Set dns server on azure bridge for ubuntu version >= 17

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:
1. This PR adds support for running CNI on ubuntu 18 or greater. Added code to configure dns server IP on azure bridge. systemd-resolved changed the way of configuring dns server on ubuntu VMs. Added code in CNI to adapt this change.
2. Moved HNS related APIs from platform package to cns/hnsclient package

Tested dnsserver changes in ubuntu 16 and ubuntu 18

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Fixes https://github.com/Azure/aks-engine/issues/875 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: network/network_linux.go [hunk 1]
```diff
@@ -139,6 +148,86 @@ func (nm *networkManager) saveIPConfig(hostIf *net.Interface, extIf *externalInt
...
+		}
+
+		return retrieved_version, err
+	}
+
+	return 0, fmt.Errorf("[net] Error getting major version")
+}
+
+func isGreaterOrEqaulUbuntuVersion(versionToMatch int) bool {
+	osInfo, err := platform.GetOSDetails()
```
Comment (sharmasushant) [inline network/network_linux.go]: print osinfo here

File: cns/hnsclient/hsnclient_linux.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package hnsclient
```
Comment (ashvindeodhar) [inline cns/hnsclient/hsnclient_linux.go]: spell check filename: hsn

File: network/network_linux.go [hunk 3]
```diff
@@ -139,6 +148,88 @@ func (nm *networkManager) saveIPConfig(hostIf *net.Interface, extIf *externalInt
...
+}
+
+func readDnsServerIP(ifName string) (string, error) {
+	cmd := fmt.Sprintf("systemd-resolve --status %s", ifName)
+	out, err := platform.ExecuteCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	log.Printf("[net] console output for above cmd: %s", out)
```
Comment (sharmasushant) [inline network/network_linux.go]: "console output for %s: %s", cmd, out)

Comment (tamilmani1989) [reply network/network_linux.go]: in previous line, we print the command..so not needed

**Pull request merged**